ipykernel/mplshim.py

```diff
@@ -0,0 +1,89 @@
+"""Import hooks to activate our inline backend when matplotlib is loaded
+"""
+
+import imp
+import sys
+
+def set_inline_backend(mpl):
+    from IPython.core.pylabtools import backends, configure_inline_support
+    from .kernelapp import IPKernelApp
+    mpl.interactive(True)
+    mpl.use(backends['inline'])
+    configure_inline_support(IPKernelApp.instance().shell, backends['inline'])
+
+class Py2MPLFinderShim(object):
+    @classmethod
+    def find_module(cls, fullname, path=None):
+        if (fullname != 'matplotlib') or (path is not None):
```

Comment:
I think the only problem is that the import hook loads our backend (and pyplot) in order to set up the config changes and post-execute hooks that the inline backends need. If we can delay that stuff (the call to `configure_inline_support()`) until the backend is actually loaded, I think it would be fine to set the backend on loading `matplotlib`, because the setting isn't committed until `matplotlib.backends` is loaded.